src/app/review/review.component.ts

```diff
@@ -70,13 +90,63 @@ export class ReviewComponent implements OnInit {
                 this.coursesReviewed = Promise.resolve(coursesReviewed);
             });
             this.questions = this.reviewService.getQuestions();
-        } else { // Other user types
+        } else if(this.userType == 2){
+            this.user = this.teacherService.getTeacher(this.userID);
+            Promise.all([
+                this.courses = this.teacherService.getCoursesByTeacher(this.userID),
+                this.reviews = this.teacherService.getReviewsByTeacher(this.userID),
+            ]).then(data => {
+                const courses = data[0];
+                const reviews = data[1];
+                const coursesToBeReviewed = [];
+                const coursesReviewed = [];
+                courses.forEach(course => {
+                    if (typeof (reviews.find(reviewedCourse => reviewedCourse.course_code === course.course_code)) === 'undefined') {
+                        coursesToBeReviewed.push(course);
+                    } else {
+                        coursesReviewed.push({
+                            course: course,
+                            review: reviews.find(reviewedCourse => reviewedCourse.course_code === course.course_code)
+                        });
+                    }
+                });
+                this.coursesToBeReviewed = Promise.resolve(coursesToBeReviewed);
+                this.coursesReviewed = Promise.resolve(coursesReviewed);
+            });
+            this.questions = this.reviewService.getQuestions();
+
+        } else if(this.userType == 3){
```

Comment:
Similar (if not the same) changes need to happen here as above with the teacher's section